lib/influxdb2/client/write_api.rb

```diff
@@ -223,6 +250,23 @@ def _worker
       end
     end
 
+    def _add_default_tags(data)
+      default_tags = @point_settings.default_tags
+
+      default_tags.each do |key, expression|
+        value = PointSettings.get_value(expression)
+        if data.is_a?(Point)
+          data.add_tag(key, value)
+        elsif data.is_a?(Hash)
+          data[:tags][key] = value
+        elsif data.respond_to? :map
+          data.map do |item|
+            _add_default_tags(item)
+          end.reject(&:nil?)
+        end
```

Comment:
The appending of tag value should be wrapped in separate method to avoid multiple evaluation of `PointSettings.get_value`.

```ruby
def _add_default_tags(data)
  default_tags = @point_settings.default_tags

  default_tags.each do |key, expression|
    value = PointSettings.get_value(expression)
    _add_default_tag(data, key, value)
  end
end

def _add_default_tag(data, key, value)
  if data.is_a?(Point)
    data.add_tag(key, value)
  elsif data.is_a?(Hash)
    data[:tags][key] = value
  elsif data.respond_to? :map
    data.map do |item|
      _add_default_tag(item, key, value)
    end.reject(&:nil?)
  end
end
```